options/options.go

```diff
@@ -79,10 +80,18 @@ type TerragruntOptions struct {
 }
 
 // Create a new TerragruntOptions object with reasonable defaults for real usage
-func NewTerragruntOptions(terragruntConfigPath string) *TerragruntOptions {
+func NewTerragruntOptions(terragruntConfigPath string) (*TerragruntOptions, error) {
 	workingDir := filepath.Dir(terragruntConfigPath)
 
-	downloadDir := filepath.Join(os.TempDir(), "terragrunt")
+	logger := util.CreateLogger("")
+
+	homedir, err := homedir.Dir()
+	if err != nil {
+		logger.Panicf("error: %v\n", err)
```

Comment:
This is a bit of a gotcha with golang: `Panicf` logs your message and then calls `panic`. So, yea, don't use it :)

Reply:
Another reason I need to do more go development...I think I knew that and forgot... :)